### DISCUSSION TITLE
Battery consuming increase without coding

### DISCUSSION THREAD

**Discussion by nam-hle** [community/community#11092] [Copilot Conversations] [answered]
Firstly, thanks to the team for bringing a new experience to the community. The AI does its job well. I wonder whether the bot runs silently after closing the IDE (I'm using Intellij) because my Macbook Pro is out of battery after nights even when just closing the machine or even sleeping. The issue no longer happens if I disabled the plugin. I'm willing to enable/disable it several times before confirming the issue. Thanks for reading :)

**Comment by jansorg** [comment, marked as answer]
@nam-hle Thanks for reporting! This is unexpected, the GitHub Copilot plugin will be stopped when you shutdown your IDE. Closing your machine while the IDE is running should put the machine into sleep mode. Can you tell if the MacBook is perhaps kept out of sleep mode even when you close the lid?

The plugin keeps a network http connection to the GitHub servers to react quickly to changes in the editor. I believe macOS has a setting to prevent sleep mode with active network connections, but this is really a very random guess.

Is there a difference in battery power when you shutdown the IDE with the Copilot plugin installed and then close the lid?

This might be related to CPU usage while using your IDE. Could you check IntelliJ's CPU usage while you're using it with Copilot, please?

**Reply by jansorg** [reply]
@nam-hle Do you think that perhaps this might be causing the battery drain? https://www.theverge.com/2022/2/10/22927670/apple-macos-update-12-2-1-battery-bluetooth-macbook-bug-ios-watchos

**Reply by nam-hle** [reply]
@jansorg Thanks for taking some time to investigate my issue. It seems about the HTTP connections are still alive after closing the lid when performing some script to detect that. I close the issue here. Let's see how better it is. 

Thanks to the whole team again, the Copilot really helps us in the brilliant ways ;)
